### PR TITLE
Feature: Add Services layer and HttpClient

### PR DESCRIPTION
**Pull request by ishakhorski-dev**
- Add Axios as HttpClient
- Add `X-Client-Id`, `X-Client-Secret`, `X-Request-Id` headers
- Add Services layer (UserSessionService, QuizSessionService)

### CODE REVIEW

File: src/plugins/AxiosPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios'
+import { v4 as uuid } from 'uuid'
+
+const axiosInstance = axios.create({})
+
+axiosInstance.interceptors.request.use((config) => {
+    config.headers.set('X-Client-Id', import.meta.env.VITE_CLIENT_ID)
```
Comment (ishakhorski-dev) [inline src/plugins/AxiosPlugin.ts:7]: @Sierrawolf87 I will suggest to use `X-Client-Id` to define from which client we are requested

File: src/plugins/AxiosPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    config.headers.set('X-Client-Secret', import.meta.env.VITE_CLIENT_SECRET)
```
Comment (ishakhorski-dev) [inline src/plugins/AxiosPlugin.ts:8]: @Sierrawolf87 I will suggest to use `X-Client-Secret` to pass client secret for more secure connection

File: src/plugins/AxiosPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    config.headers.set('X-Request-Id', uuid())
```
Comment (ishakhorski-dev) [inline src/plugins/AxiosPlugin.ts:9]: @Sierrawolf87 I will suggest to use `X-Request-Id` to define request id for better debug process

File: src/plugins/AxiosPlugin.ts (hunk 1 above)
Comment (Sierrawolf87) [reply src/plugins/AxiosPlugin.ts:7]: Will be implemented in the next version

File: src/plugins/AxiosPlugin.ts (hunk 2 above)
Comment (Sierrawolf87) [reply src/plugins/AxiosPlugin.ts:8]: Will be implemented in the next version

File: src/plugins/AxiosPlugin.ts (hunk 3 above)
Comment (Sierrawolf87) [reply src/plugins/AxiosPlugin.ts:9]: Will be implemented in the next version

File: src/services/BaseService.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { AxiosHttpClient } from './http/AxiosHttpClient'
+import { AxiosPlugin } from '@/plugins'
+
+const HttpClientInstance = new AxiosHttpClient(AxiosPlugin)
+
+export default class BaseService {
+    protected http: HttpClient = HttpClientInstance
+    protected baseUrl: string = import.meta.env.VITE_API_URL
+
+    get<T = unknown>(path: string) {
```
Comment (Sierrawolf87) [inline src/services/BaseService.ts:11]: Maybe we need additional options for all http requests?

Comment (ishakhorski-dev) [reply src/services/BaseService.ts]: You are right, I'll update PR soon!

**Review by Sierrawolf87 — APPROVED** [review]

**Pull request merged**